QDK Release Notes Aug 2020 (#1019)
MicrosoftDocs/quantum-docs-pr: articles/resources/relnotes.md

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

## Version X.XX.XXXXXXXX

*Release date: August 25th, 2020*

This release contains the following:

> - General bug fixes.

Reviewer: Could we move "General bug fixes" to the end? Also I think we should add the standard pointers to GitHub at the end:

```
See the full list of closed PRs for [libraries](https://github.com/Microsoft/QuantumLibraries/pulls?q=is%3Apr+is%3Aclosed), [compiler](https://github.com/microsoft/qsharp-compiler/pulls?q=is%3Apr+is%3Aclosed), [runtime](https://github.com/microsoft/qsharp-runtime/pulls?q=is%3Apr+is%3Aclosed), [samples](https://github.com/Microsoft/Quantum/pulls?q=is%3Apr+is%3Aclosed), [IQ#](https://github.com/microsoft/iqsharp/pulls?q=is%3Apr+is%3Aclosed) and [Katas](https://github.com/microsoft/QuantumKatas/pulls?q=is%3Apr+is%3Aclosed).  
```

> - Fixed bug in M.Q.Arithmetic.RippleCarryAdderCDKM.

Reviewer 2: ```suggestion
- Fixed bug in Microsoft.Quantum.Arithmetic.RippleCarryAdderCDKM.
```

> - Fixed bug in M.Q.Arithmetic.RippleCarryAdderCDKM.

Reviewer 3: We will have a generic line for bug fixes in the release notes, do we want to mention this bug fix in particular? @msoeken @KittyYeungQ

- Improved Microsoft.Quantum.Random.

> - Improved Microsoft.Quantum.Diagnostics.

Reviewer: I think these could use some more detail - what was improved?

> - Fixed intermittent QDK build failures due to access violation: "Test host process crashed".

Reviewer: I think this one should be phrased differently to reflect the potential benefit to users, rather than the impact on our internal build pipeline. @swernli do you have a suggestion here?

Reviewer 3: If this was part of the bug bash, we agreed we will just mention bug fixes in general in release notes

> - Added IntelliSense support when using Python as a driver.

Reviewer: ```suggestion
- Added IntelliSense support in VS/VS Code extensions for Q# files in folders without `.csproj` files (for example, when working with `.qs` files directly from Python).
- New [`%project` magic command](xref:microsoft.quantum.iqsharp.magic-ref.project) in IQ# and [`qsharp.projects` API](https://docs.microsoft.com/python/qsharp/qsharp.projects.projects) in Python to support references to Q# projects outside the current workspace folder. See [iqsharp#277](https://github.com/microsoft/iqsharp/issues/277) for the current limitations of this feature.
- Support for automatically loading `.csproj` files for IQ#/Python hosts, which allows external project or package references to be loaded at initialization time.
   - This behavior can be enabled by adding a `.csproj` file (in the folder containing the `.py` or `.ipynb` file) which references `Microsoft.Quantum.Sdk` and contains the property `<IQSharpLoadAutomatically>true</IQSharpLoadAutomatically>`.
```

> 

Reviewer 4: ```suggestion
- Improved Microsoft.Quantum.Samples
    - Added ErrorCorrection.Syndrome sample
    - Separated SimpleAlgorithms into different files
    - Added tunable coupling to SimpleIsing
```

> - Improved Microsoft.Quantum.Diagnostics

Reviewer 2: ```suggestion
- Improved Microsoft.Quantum.Diagnostics.
```

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

> ## Version X.XX.XXXXXXXX

Reviewer: ```suggestion
## Version 0.12.20082406
```

- Improved [Microsoft.Quantum.Diagnostics namespace](xref:microsoft.quantum.diagnostics) with new [`DumpOperation` operation](xref:microsoft.quantum.diagnostics.dumpoperation), and new operations for restricting qubit allocation and oracle calls. ([QuantumLibraries#302](https://github.com/microsoft/QuantumLibraries/pull/302))
- Added IntelliSense support in VS/VS Code extensions for Q# files in folders without `.csproj` files (for example, when working with `.qs` files directly from Python).
- New [`%project` magic command](xref:microsoft.quantum.iqsharp.magic-ref.project) in IQ# and [`qsharp.projects` API](https://docs.microsoft.com/python/qsharp/qsharp.projects.projects) in Python to support references to Q# projects outside the current workspace folder. See [iqsharp#277](https://github.com/microsoft/iqsharp/issues/277) for the current limitations of this feature.
- Added ErrorCorrection.Syndrome sample
- Added tunable coupling to SimpleIsing
- Updated HiddenShift sample
- Added sample for solving Sudoku with Grover's algorithm (external contribution)

> - General bug fixes. See the full list of closed PRs for [libraries](https://github.com/Microsoft/QuantumLibraries/pulls?q=is%3Apr+is%3Aclosed), [compiler](https://github.com/microsoft/qsharp-compiler/pulls?q=is%3Apr+is%3Aclosed), [runtime](https://github.com/microsoft/qsharp-runtime/pulls?q=is%3Apr+is%3Aclosed), [samples](https://github.com/Microsoft/Quantum/pulls?q=is%3Apr+is%3Aclosed), [IQ#](https://github.com/microsoft/iqsharp/pulls?q=is%3Apr+is%3Aclosed) and [Katas](https://github.com/microsoft/QuantumKatas/pulls?q=is%3Apr+is%3Aclosed).  

Reviewer: ```suggestion
- General bug fixes.

See the full list of closed PRs for [libraries](https://github.com/Microsoft/QuantumLibraries/pulls?q=is%3Apr+is%3Aclosed), [compiler](https://github.com/microsoft/qsharp-compiler/pulls?q=is%3Apr+is%3Aclosed), [runtime](https://github.com/microsoft/qsharp-runtime/pulls?q=is%3Apr+is%3Aclosed), [samples](https://github.com/Microsoft/Quantum/pulls?q=is%3Apr+is%3Aclosed), [IQ#](https://github.com/microsoft/iqsharp/pulls?q=is%3Apr+is%3Aclosed) and [Katas](https://github.com/microsoft/QuantumKatas/pulls?q=is%3Apr+is%3Aclosed).  
```

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

## Version X.XX.XXXXXXXX

*Release date: August 25th, 2020*

This release contains the following:

- General bug fixes.
- Improved Microsoft.Quantum.Random.
- Improved Microsoft.Quantum.Diagnostics.
- Added IntelliSense support in VS/VS Code extensions for Q# files in folders without `.csproj` files (for example, when working with `.qs` files directly from Python).
- New [`%project` magic command](xref:microsoft.quantum.iqsharp.magic-ref.project) in IQ# and [`qsharp.projects` API](https://docs.microsoft.com/python/qsharp/qsharp.projects.projects) in Python to support references to Q# projects outside the current workspace folder. See [iqsharp#277](https://github.com/microsoft/iqsharp/issues/277) for the current limitations of this feature.

> - Support for automatically loading `.csproj` files for IQ#/Python hosts, which allows external project or package references to be loaded at initialization time. See the guide for using [Q# with Python and Jupyter Notebooks](xref:microsoft.quantum.guide.host-programs) for more details.

Reviewer: We decided to postpone this feature until the next release.
```suggestion
```

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update).

> ## Version 0.12.20082406

Reviewer: ```suggestion
## Version 0.12.20082513
```

*Release date: August 25th, 2020*

This release contains the following:

- New [Microsoft.Quantum.Random namespace](xref:microsoft.quantum.random), providing a more convenient way to sample random values from within Q# programs. ([QuantumLibraries#311](https://github.com/microsoft/QuantumLibraries/pull/311), [qsharp-runtime#328](https://github.com/microsoft/qsharp-runtime/pull/328))
- Improved [Microsoft.Quantum.Diagnostics namespace](xref:microsoft.quantum.diagnostics) with new [`DumpOperation` operation](xref:microsoft.quantum.diagnostics.dumpoperation), and new operations for restricting qubit allocation and oracle calls. ([QuantumLibraries#302](https://github.com/microsoft/QuantumLibraries/pull/302))

> - Added IntelliSense support in VS/VS Code extensions for Q# files in folders without `.csproj` files (for example, when working with `.qs` files directly from Python).
> - New [`%project` magic command](xref:microsoft.quantum.iqsharp.magic-ref.project) in IQ# and [`qsharp.projects` API](https://docs.microsoft.com/python/qsharp/qsharp.projects.projects) in Python to support references to Q# projects outside the current workspace folder. See [iqsharp#277](https://github.com/microsoft/iqsharp/issues/277) for the current limitations of this feature.

Reviewer: I'm sorry, my last suggestion removed the wrong line here. Adding that one back and removing the one I really wanted to remove:
```suggestion
- New [`%project` magic command](xref:microsoft.quantum.iqsharp.magic-ref.project) in IQ# and [`qsharp.projects` API](https://docs.microsoft.com/python/qsharp/qsharp.projects.projects) in Python to support references to Q# projects outside the current workspace folder. See [iqsharp#277](https://github.com/microsoft/iqsharp/issues/277) for the current limitations of this feature.
- Support for automatically loading `.csproj` files for IQ#/Python hosts, which allows external project or package references to be loaded at initialization time. See the guide for using [Q# with Python and Jupyter Notebooks](xref:microsoft.quantum.guide.host-programs) for more details.
```

> - Added ErrorCorrection.Syndrome sample
> - Added tunable coupling to SimpleIsing
> - Updated HiddenShift sample

Reviewer: ```suggestion
- Added ErrorCorrection.Syndrome sample.
- Added tunable coupling to SimpleIsing.
- Updated HiddenShift sample.
```
